src/transformers/modeling_utils.py

```diff
@@ -2700,7 +2701,7 @@ def from_pretrained(
             quantization_method_from_args == QuantizationMethod.GPTQ
             or quantization_method_from_config == QuantizationMethod.GPTQ
         ):
-            if not torch.cuda.is_available():
+            if not load_gptq_on_cpu and not torch.cuda.is_available():
```

Comment:
```suggestion
            gptq_supports_cpu = version.parse(importlib.metadata.version("auto-gptq")) > version.parse("0.4.2")
            
            if not gptq_supports_cpu and not torch.cuda.is_available():
```

I think we can support that in a more elegant way, can we create an attribute `gptq_supports_cpu = version.parse(importlib.metadata.version("auto-gptq")) > version.parse("0.4.2")` similary as :https://github.com/huggingface/transformers/blob/6df9179c1c5b19674ce0a4b82d311ef833c02447/src/transformers/modeling_utils.py#L2496

Then we can safely remove `load_gptq_on_cpu` arg safely from `from_pretrained`. We should also force-set `disable-exllama` to `True` in that case

Reply:
Don't forget also to change this: https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2721 to print the warning only in case `not gptq_supports_cpu`